sdk/python/lib/pulumi/runtime/resource.py

```diff
@@ -170,9 +170,11 @@ def read_resource(res: 'Resource', ty: str, name: str, props: 'Inputs', opts: Op
     #
     # Note that we technically already have the ID (opts.id), but it's more consistent with the rest
     # of the model to resolve it asynchronously along with all of the other resources.
-    resolve_value = asyncio.Future()
-    resolve_perform_apply = asyncio.Future()
-    resolve_secret = asyncio.Future()
+    res = cast(CustomResource, res) # pylint: disable=used-before-assignment
```

Comment:
Feels like we should actually fix this - either make `read_resource` accept only a `CustomResource`, or do an `isinstance` check and fail.  This seems like it could easily legitimately not be true.

Reply:
made the change to the method signature.